### DISCUSSION TITLE
Many to Many relationship ordering

### DISCUSSION THREAD

**Discussion by un-focused** [prisma/prisma#8489] [Q&A]
Do I need a link table in between two models to allow for keeping the order between them?

For instance:
```prisma
model A {
  id                 Int       @id @default(autoincrement())
  b                  B[]
}

model B {
  id                 Int       @id @default(autoincrement())
  a                  A[]
}
```

How can I store B models in A so that it is kept in order? I've used link models before where it contains an index property but it seems Prisma already supports this for lists/array?

[Scalar Lists](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-fields/working-with-scalar-lists-arrays)
[List Order](https://v1.prisma.io/forum/t/list-array-order-guaranteed/2235)
[Many to Many](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/many-to-many-relations)

**Comment by ryands17** [comment]
@un-focused 👋 

You would need to create a separate relation table to have an order field also known as [explicit many-to-many relation](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/many-to-many-relations#explicit-many-to-many-relations). So something like this:

```prisma
model A {
  id Int  @id @default(autoincrement())
  b  AB[]
}

model B {
  id Int  @id @default(autoincrement())
  a  AB[]
}

model AB {
  id  Int  @id @default(autoincrement())
  order Int
  a   A?   @relation(fields: [aId], references: [id])
  aId Int?
  b   B?   @relation(fields: [bId], references: [id])
  bId Int?
}
```

**Reply by ryands17** [reply]
@un-focused let me know if that answers your question 🙂

**Reply by un-focused** [reply]
Not quite, I saw explicit many-to-many relations and uses them already. I'm wondering if there is a way to do keep order without an explicit many-to-many relation because I setup an implicit relation in production and so switching would require a migration

**Reply by un-focused** [reply]
Is there a way to migrate easily from implicit to explicit?

**Reply by ryands17** [reply]
> Not quite, I saw explicit many-to-many relations and uses them already

Could you share an example of this? Explicit relations have to setup manually so even the order needs to be added. It's not available automatically.

> Is there a way to migrate easily from implicit to explicit?

You would need to create the explicit table first along with relations, add the values from the implicit table to the explicit one.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
